modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/TpchQueryPlannerTest.java

```diff
@@ -17,75 +17,44 @@
 
 package org.apache.ignite.internal.sql.engine.planner;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.apache.ignite.internal.sql.engine.planner.AbstractTpcQueryPlannerTest.TpcSuiteInfo;
 
-import com.google.common.io.CharStreams;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.io.UncheckedIOException;
-import java.nio.charset.StandardCharsets;
-import org.apache.calcite.plan.RelOptUtil;
-import org.apache.calcite.sql.SqlExplainLevel;
-import org.apache.ignite.internal.sql.engine.framework.TestBuilders;
-import org.apache.ignite.internal.sql.engine.framework.TestCluster;
-import org.apache.ignite.internal.sql.engine.framework.TestNode;
-import org.apache.ignite.internal.sql.engine.prepare.MultiStepPlan;
-import org.apache.ignite.internal.sql.engine.util.Cloner;
-import org.apache.ignite.internal.sql.engine.util.Commons;
 import org.apache.ignite.internal.sql.engine.util.tpch.TpchHelper;
 import org.apache.ignite.internal.sql.engine.util.tpch.TpchTables;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 
 /**
  * Tests ensures a planner generates optimal plan for TPC-H queries.
  *
  * @see org.apache.ignite.internal.sql.engine.benchmarks.TpchParseBenchmark
  */
 // TODO https://issues.apache.org/jira/browse/IGNITE-21986 validate other query plans and make test parameterized.
-public class TpchQueryPlannerTest extends AbstractPlannerTest {
-    private static TestCluster CLUSTER;
-
-    @BeforeAll
-    static void startCluster() {
-        CLUSTER = TestBuilders.cluster().nodes("N1").build();
-        CLUSTER.start();
-
-        TestNode node = CLUSTER.node("N1");
-
-        node.initSchema(TpchTables.LINEITEM.ddlScript());
+@TpcSuiteInfo(
+        tables = TpchTables.class,
+        queryLoader = "getQueryString",
+        planLoader = "getQueryPlan"
+)
+public class TpchQueryPlannerTest extends AbstractTpcQueryPlannerTest {
+    @ParameterizedTest
+    // Query #7 contains disjunctive predicate that switches its part from run to run, making
```

Comment:
I think the reason is we use. HashSet/HashMap somewhere.
E.g.  `Set<CorrelationId> correlationIds = new HashSet<>();` in CorrelatedNestedLoopJoinRule

Reply:
Discussed. We agreed the real reason is Calcite RexUtils code generates unstable conditions order.
BWT, on our side, we should use LinkedHashSet/Map to preserve stable digest/plan/toString results.
Let's create a ticket.

Reply:
https://issues.apache.org/jira/browse/IGNITE-24195